packages/react-native/src/theme/hooks/usePressableOverrideStyle.ts

```diff
@@ -0,0 +1,34 @@
+import { useCallback } from 'react';
+import {
+  PressableProps,
+  PressableStateCallbackType,
+  StyleProp,
+  ViewStyle,
+} from 'react-native';
+
+export interface UsePressableOverrideStyleProps
+  extends Pick<PressableProps, 'style'> {
+  themedStyle?: StyleProp<ViewStyle>;
+  pressedStyle?: StyleProp<ViewStyle>;
+}
+
+export const usePressableOverrideStyle = ({
+  style,
+  themedStyle,
```

Comment:
Since `themedStyle` is passed through the hook and returned without having an impact on the functionality of the hook, it doesn't feel like we need it here. Ultimately this hook reads to me as a generic utility for handling a `Pressable['style']` passed as a prop to a component and applying that component's `pressed` state style. 

Additionally my thinking is that removing `themedStyle` will also make this hook easier to understand. In this commit we are passing theme related styles for `pressedStyle` while also having `themedStyle` which may be confusing when referencing its usage by future authors. For example, this is from the usage in the `Tabs` component:
```tsx
  const pressableStyle = usePressableOverrideStyle({
    style,
    themedStyle: containerStyle,
    pressedStyle: themedStyle.pressed,
  });
 ```